packages/components/src/internal/components/search/SampleFinderSection.tsx

```diff
@@ -59,50 +74,86 @@ export const SampleFinderHeaderButtons: FC<SampleFinderHeaderProps> = memo(props
     );
 });
 
+
+function getLocalStorageKey(): string {
+    return getPrimaryAppProperties().productId + ActionURL.getContainer() + "-SampleFinder";
+}
+
 export const SampleFinderSection: FC<Props> = memo(props => {
     const { parentEntityDataTypes, ...gridProps } = props;
 
+    const [filterChangeCounter, setFilterChangeCounter] = useState<number>(0);
     const [chosenEntityType, setChosenEntityType] = useState<EntityDataType>(undefined);
-    const [filterCards, setFilterCards] = useState<FilterCardProps[]>([]);
+    const [filters, setFilters] = useState<FilterProps[]>([]);
+
+    useEffect(() => {
+        const finderSessionDataStr = sessionStorage.getItem(getLocalStorageKey());
+        if (finderSessionDataStr) {
+            const finderSessionData = JSON.parse(finderSessionDataStr);
+            if (finderSessionData.filters) {
+                setFilters(finderSessionData.filters);
+            }
+            if (finderSessionData.filterChangeCounter !== undefined) {
+                setFilterChangeCounter(finderSessionData.filterChangeCounter);
+            }
+        }
+    }, []);
+
+    const getSelectionKeyPrefix = (): string => {
+        return 'sampleFinder-' + filterChangeCounter;
+    };
+
+    const updateSessionStorage = (filterChangeCounter: number, filters: FilterProps[]) => {
```

Comment:
it looks like `setFilters` and `setFilterChangeCounter` are both always called before this is called, so should those calls just move into this function instead (so we don't forget one)?